charmtools/build/tactics.py

```diff
@@ -378,9 +399,51 @@ class MetadataYAML(YAMLTactic):
                  "description", "tags",
                  "requires", "provides", "peers"]
 
+    STORAGE_BINDING = HOOK_TEMPLATE
+
+    def __init__(self, *args, **kwargs):
+        super(MetadataYAML, self).__init__(*args, **kwargs)
+        self.storage = {}
+
+    def read(self):
+        if not self._read:
+            super(MetadataYAML, self).read()
+            names = self.data.get('storage', {}).keys()
+            self.storage = {name: self.current.url for name in names}
+
+    def combine(self, existing):
+        super(MetadataYAML, self).combine(existing)
+        self.storage.update(existing.storage)
+        return self
+
+    def __call__(self):
+        super(MetadataYAML, self).__call__()
+        for name in self.storage.keys():
+            for hook in ['joined', 'changed', 'broken', 'departed']:
+                target = self._target / "hooks" / "{}-relation-{}".format(
+                    name, hook)
+                if target.exists():
+                    continue
+                target.parent.makedirs_p()
+                target.write_text(self.STORAGE_BINDING.format(name))
```

Comment:
Storage does not follow `{storage-name}-relation-{hook}` but rather `{storage-name}-attached` and `{storage-name}-detaching}` [ref](https://jujucharms.com/docs/1.25/storage#implementing-hooks). I don't see how that would be created form this.